docs/proposals/runtime-client-server.md

```diff
@@ -0,0 +1,180 @@
+    // ListPodSandbox returns a list of SandBox.
+    rpc ListPodSandbox(ListPodSandboxRequest) returns (ListPodSandboxResponse) {}
+    // PortForward copies the data from the port in the PodSandBox to the stream.
+    rpc PortForward(stream PortForwardRequest) returns (stream PortForwardResponse) {}
+
+    // CreateContainer creates a new container in specified PodSandbox
+    rpc CreateContainer(CreateContainerRequest) returns (CreateContainerResponse) {}
+    // StartContainer starts the container.
+    rpc StartContainer(StartContainerRequest) returns (StartContainerResponse) {}
+    // StopContainer stops a running container with a grace period (i.e., timeout).
+    rpc StopContainer(StopContainerRequest) returns (StopContainerResponse) {}
+    // RemoveContainer removes the container.
+    rpc RemoveContainer(RemoveContainerRequest) returns (RemoveContainerResponse) {}
+    // ListContainers lists all containers by filters.
+    rpc ListContainers(ListContainersRequest) returns (ListContainersResponse) {}
+    // ContainerStatus returns the container status.
+    rpc ContainerStatus(ContainerStatusRequest) returns (ContainerStatusResponse) {}
+    // Exec execute a command in the container.
+    rpc Exec(stream ExecRequest) returns (stream ExecResponse) {}
+}
+
+// Image service defines the public APIs for managing images
+service ImageService {
+    // ListImages lists existing images.
+    rpc ListImages(ListImagesRequest) returns (ListImagesResponse) {}
+    // ImageStatus returns the status of the image.
+    rpc ImageStatus(ImageStatusRequest) returns (ImageStatusResponse) {}
+    // PullImage pulls a image with authentication config.
+    rpc PullImage(PullImageRequest) returns (PullImageResponse) {}
+    // RemoveImage removes the image.
+    rpc RemoveImage(RemoveImageRequest) returns (RemoveImageResponse) {}
+}
+```
+
+Note that some type in the proto is already defined at [Container runtime interface/integration](https://github.com/kubernetes/kubernetes/pull/25899).
+We should decide how to integrate the types in [#25899](https://github.com/kubernetes/kubernetes/pull/25899) with gRPC services:
+
+* Auto-generate those types into proto by `go2idl`
+  - Pros: trace type changes automatically, no need type conversions
+  - Cons: type change may break existing API implementations, should process attributes order carefully so as not to break generated protos
+* Embed those types as raw proto definitions
+  - Pros: duplicate definition and conversion of same types, hard to track type changes automatically
+  - Cons: decouple type definitions, more easy to manage gRPC API versions
+
+As @yujuhong suggested, we decide to embed all those types in `api.proto` directly for fast iterations.
+
+## Logging
+
+The logging of container runtime is discussed at [#24677](https://github.com/kubernetes/kubernetes/issues/24677) and [#27154](https://github.com/kubernetes/kubernetes/issues/27154).
```

Comment:
There are a couple of issues in discussion besides logging (e.g., attach, portforward). Either list all of them or exclude the discussion from this proposal, since the proposal focuses more on *how* to implement the client-server architecture.